bundles/automation/org.eclipse.smarthome.automation.module.core/src/main/java/org/eclipse/smarthome/automation/module/core/handler/RuleEnableHandler.java

```diff
@@ -0,0 +1,127 @@
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceReference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class is a handler for RuleEnablementAction module type.
+ * It enables or disables the rules which's UIDs are passed by the 'ruleUIDs' property.
+ * !!! If a rule's status is NOT_INITIALIZED that rule can't be enabled. !!!
+ *
+ * <pre>
+ *Example:
+ *
+ *"id": "RuleAction",
+ *"type": "RuleEnablementAction",
+ *"configuration": {
+ *     "enable": true,
+ *     "ruleUIDs": ["UID1", "UID2", "UID3"]
+ * }
+ * </pre>
+ *
+ * @author Plamen Peev
+ *
+ */
+public class RuleEnableHandler extends BaseModuleHandler<Action> implements ActionHandler {
+
+    /**
+     * This filed contains the type of this handler so it can be recognized from the factory.
+     */
+    public final static String UID = "RuleEnablementAction";
+
+    /**
+     * This field is a key to the 'enable' property of the {@link Action}.
+     */
+    private final static String ENABLE_KEY = "enable";
+
+    /**
+     * This field is a key to the 'rulesUIDs' property of the {@link Action}.
+     */
+    private final static String RULE_UIDS_KEY = "ruleUIDs";
+
+    /**
+     * This logger is used to log warning message if at some point {@link RuleRegistry} service becomes unavailable.
+     */
+    private final static Logger logger;
+
+    /**
+     * A reference to the bundle context of the framework. Will be used to get the {@link RuleRegistry} service.
+     */
+    private final BundleContext bundleContext;
+
+    /**
+     * This field stores the UIDs of the rules to which the action will be applied.
+     */
+    private final List<String> UIDs;
+
+    /**
+     * This field stores the value for the setEnabled() method of {@link RuleRegistry}.
+     */
+    private final boolean enable;
+
+    static {
+        logger = LoggerFactory.getLogger(RuleEnableHandler.class);
+    }
+
+    @SuppressWarnings("unchecked")
+    public RuleEnableHandler(final Action module, final BundleContext context) {
+        super(module);
+        final Configuration config = module.getConfiguration();
+        if (config == null) {
+            throw new IllegalArgumentException("'Configuration' can not be null.");
+        }
+
+        final Boolean enable = (Boolean) config.get(ENABLE_KEY);
+        if (enable == null) {
+            throw new IllegalArgumentException("'enable' property can not be null.");
+        }
+        this.enable = enable.booleanValue();
+
+        UIDs = (List<String>) config.get(RULE_UIDS_KEY);
+        if (UIDs == null) {
+            throw new IllegalArgumentException("'ruleUIDs' property can not be null.");
+        }
+
+        this.bundleContext = context;
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public Map<String, Object> execute(Map<String, ?> context) {
+        final ServiceReference<RuleRegistry> reference = (ServiceReference<RuleRegistry>) bundleContext
```

Comment:
Yes I agree that internal trackers are not good option, but only this MT depends from RuleRegistry. If we add static dependency to RuleRegistry (which is different from RuleEngine and RuleEngine has to work without RuleRegistry), we have to remove all core MT defined by this  Factory which can be used by the RuleEngine.
May be this MT has to be defined by separate component with its own factory.